ignite/engine/engine.py

```diff
@@ -695,25 +695,32 @@ def _internal_run(self) -> State:
                     self._setup_engine()
 
                 time_taken = self._run_once_on_dataset()
+                # time is available for handlers but must be update after fire
+                self.state.times[Events.EPOCH_COMPLETED.name] = time_taken
+                fire_start_time = time.time()
```

Comment:
nit: `fire_start_time` -> `handlers_start_time` or just `start_time` ?

Reply:
`start_time` already in use but `handlers_start_time` is nice and explicit !